sensors.py

```diff
@@ -104,14 +113,14 @@ def get_reading(self, sensor_type: SensorType) -> int | float:
         with self._lock:
             try:
                 self.update()
-            except:
-                logging.exception("Couldn't get a reading")
-                raise SensorReadingError
-                
+            except Exception as exc:
+                raise SensorReadingError from exc
+
             if sensor_type not in self.pm:
                 raise WrongSensorType
-            if self.pm[sensor_type] is None:
+            tmp = self.pm[sensor_type] 
+            if tmp is None:
```

Comment:
`if not`?

Reply:
tmp can be None and int and I check for None not for 0(it's a valid number).